src/rabbit_exchange_type_consistent_hash.erl

```diff
@@ -159,9 +185,34 @@ find_numbers(Source, N, Acc) ->
 
 hash(undefined, #basic_message { routing_keys = Routes }) ->
     Routes;
-hash({longstr, Header}, #basic_message { content = Content }) ->
+hash({header, Header}, #basic_message { content = Content }) ->
     Headers = rabbit_basic:extract_headers(Content),
     case Headers of
         undefined -> undefined;
         _         -> rabbit_misc:table_lookup(Headers, Header)
+    end;
+hash({property, Property}, #basic_message {content = Content }) ->
+    #content{properties = #'P_basic'{correlation_id = CorrId, message_id = MsgId }} =
+        rabbit_binary_parser:ensure_content_decoded(Content),
+    case Property of
+      <<"correlation_id">> -> CorrId;
+      <<"message_id">> -> MsgId
+    end.
+
+hash_args(Args) ->
+    Header = case rabbit_misc:table_lookup(Args, <<"hash-header">>) of
```

Comment:
here you can move the `case` to the next line, and then indent the whole case block